### ISSUE TITLE
[Question] Reproduced acc but not F1 on MOSEI; ablations (w/o CAFD / w/o MAR) show only small changes — are my settings correct?

### ISSUE THREAD

**Issue by XStiger5939** [YetZzzzzz/CMAD#5]
Thanks for the great work! I reproduced the full model on MOSEI but with a noticeable F1
gap, and my two ablations changed the results only slightly. I'd like to confirm whether
my ablation settings are correct and whether these results are expected.

**Setup**
- Dataset: CMU-MOSEI
- Command (full model):
  `python3 CMAD_sentiment/Student_Model/stu_config_mosei.py --dataset=mosei --begin_epoch=20 --delta=0.1 --gamma=1 --temperature=7.0 --n_epochs=80 --train_batch_size=128`
- Single run, seed=5576 (the code default).
- "Best" below = per-metric maximum over the 80-epoch run.

**1. Full model vs paper (best)**

|             | avg_acc | avg_f1 |
|-------------|---------|--------|
| Paper       | 77.0    | 76.4   |
| My best     | 76.94   | 75.82  |
| Δ           | -0.06   | **-0.58** |

Acc reproduces well, but F1 is ~0.6 lower even at the best epoch. Is this within expected
single-seed variance, or should I check something (seed / hyperparameters / eval protocol)?

**2. Ablation results (best)**

| Setting                    | avg_acc | avg_f1 |
|----------------------------|---------|--------|
| Full (delta=0.1, begin=20) | 76.94   | 75.82  |
| w/o CAFD (delta=0)         | 76.54   | 75.70  |
| w/o MAR (begin_epoch=81)   | 77.00   | 75.66  |

Paper ablation (for comparison):
| Setting   | avg_acc | avg_f1 |
|-----------|---------|--------|
| Full      | 77.0    | 76.4   |
| w/o CAFD  | 78.8     |   76.8  |
| w/o MAR   | 78.9     | 76.4    |

Removing CAFD drops acc/f1 by only ~0.4/0.1, and removing MAR barely changes the best
numbers (acc even slightly higher). This looks weaker than the paper's ablation table.

**3. A per-combination observation (MAR)**

At the same epoch (74), MAR's effect per combination (Full − w/o MAR):

| Combination | Δ acc | Δ f1 |
|-------------|-------|------|
| [1,1,1]     | +0.05 | +0.16 |
| [1,0,0]     | +0.36 | +0.28 |
| [1,1,0]     | +0.16 | +0.12 |
| [1,0,1]     | +0.27 | +0.40 |
| [0,0,1]     | +0.50 | +1.09 |
| [0,1,1]     | +0.58 | +0.64 |
| [0,1,0] audio-only | **+2.90** | **-3.03** |

MAR helps acc on every combination (especially audio-only, +2.90), which matches the
paper's claim. But on audio-only, MAR improves acc by +2.90 while *hurting* F1 by -3.03,
which cancels out the F1 gains on the other combinations in the average. Is this
acc↑ / F1↓ behavior on the hardest combination expected?

**Questions**
1. Are my ablation settings correct?
   - w/o CAFD: `--delta=0`
   - w/o MAR: `--begin_epoch=81` (so `combination_importance` never takes effect)
2. Is `delta=0.1` the value used in the paper? I noticed the CAFD term contributes only
   `delta * wei_mse_loss ≈ 0.1 * 0.34 ≈ 0.03` out of a total loss ≈ 25 (~0.1%), so it is
   almost drowned out — is this intended?
3. Why does MAR show opposite acc/F1 effects on the audio-only combination?
4. What seeds and selection criterion does the paper use for MOSEI? (I only ran seed 5576,
   and "best" acc/F1 here come from *different* epochs due to an acc/F1 tradeoff.)

Happy to share full training logs. Thanks!
